src/main/java/org/terasology/launcher/game/GameManager.java

```diff
@@ -132,29 +136,91 @@ public Path getInstallDirectory(GameIdentifier id) {
+        } catch (IOException e) {
+            logger.warn("Error while scanning installation directory {}:", installDirectory, e);
+            return;
         }
         Platform.runLater(() -> installedGames.addAll(localGames));
     }
+
+    private static GameIdentifier getInstalledVersion(Path versionDirectory) {
+        Profile profile;
+        Build build;
+        var parts = versionDirectory.getNameCount();
+        try {
+            profile = Profile.valueOf(versionDirectory.getName(parts - 3).toString());
+            build = Build.valueOf(versionDirectory.getName(parts - 2).toString());
+        } catch (IllegalArgumentException e) {
+            logger.debug("Directory does not match expected profile/build names: {}", versionDirectory, e);
+            return null;
+        }
+        return getInstalledVersion(profile, build, versionDirectory);
+    }
+
+    private static GameIdentifier getInstalledVersion(Profile profile, Build build, Path versionDirectory) {
+        Path engineJar;
+
+        try (var jarMatches = Files.find(versionDirectory, 3, GameManager::matchEngineJar)) {
+            var matches = jarMatches.collect(Collectors.toUnmodifiableSet());
+            if (matches.isEmpty()) {
+                logger.warn("Could not find engine jar in {}", versionDirectory);
+                return null;
+            } else if (matches.size() > 1) {
+                logger.warn("Ambiguous results while looking for engine jar in {}: {}", versionDirectory, matches);
+                return null;
+            }
+            engineJar = matches.iterator().next();
+        } catch (IOException e) {
+            logger.error("Error while looking for engine jar in {}", versionDirectory, e);
+            return null;
+        }
+
+        Properties versionInfo;
+        try {
+            versionInfo = getVersionPropertiesFromJar(engineJar);
+        } catch (IOException e) {
+            logger.error("Error while looking for version in {}.", engineJar, e);
+            return null;
+        }
+
+        return GameIdentifier.fromVersionInfo(versionInfo, build, profile).orElse(null);
+    }
+
+    private static boolean matchEngineJar(Path path, BasicFileAttributes basicFileAttributes) {
+        // Are there path-matching utilities to simplify this?
+        final var libPaths = Set.of(Path.of("lib"), Path.of("libs"));
+
+        var parent = path.getParent();
+        var file = path.getFileName().toString();
+        return Files.isDirectory(parent)
+                && libPaths.contains(parent.getFileName())
+                && file.endsWith(".jar")
+                && file.startsWith("engine");
+    }
+
+    private static Properties getVersionPropertiesFromJar(Path jarLocation) throws IOException {
+        try (var jar = new JarFile(jarLocation.toFile())) {
+            var versionEntry = jar.stream().filter(entry ->
+                    entry.getName().equals("versionInfo.properties")  // FIXME: use const
```

Comment:
I've got some older installations in my testing workspace, and this will throw an error, even though the version info file is found.

```
10:14:56.283 [Launcher init thread] ERROR o.t.launcher.game.GameManager - Error while looking for version in /home/skaldarnar/Code/movingblocks/terasologylauncher/build/classes/games/OMEGA/NIGHTLY/alpha-20+150/libs/engine-4.4.0-SNAPSHOT.jar.
java.io.FileNotFoundException: Found no versionInfo.properties in /home/skaldarnar/Code/movingblocks/terasologylauncher/build/classes/games/OMEGA/NIGHTLY/alpha-20+150/libs/engine-4.4.0-SNAPSHOT.jar
        at org.terasology.launcher.game.GameManager.getVersionPropertiesFromJar(GameManager.java:217)
        at org.terasology.launcher.game.GameManager.getInstalledVersion(GameManager.java:190)
        at org.terasology.launcher.game.GameManager.getInstalledVersion(GameManager.java:167)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at org.terasology.launcher.game.GameManager.scanInstallationDir(GameManager.java:148)
        at org.terasology.launcher.game.GameManager.<init>(GameManager.java:49)
        at org.terasology.launcher.LauncherInitTask.call(LauncherInitTask.java:95)
        at org.terasology.launcher.LauncherInitTask.call(LauncherInitTask.java:42)
        at javafx.graphics/javafx.concurrent.Task$TaskCallable.call(Task.java:1425)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.lang.Thread.run(Thread.java:832)
```

There is a `versionInfo.properties` file in `alpha-20+150/libs/engine-4.4.0-SNAPSHOT.jar`, located under `/org/terasology/version/`. As the entry names contain the full path to the file it is not matched by `entry.getName().equals("versionInfo.properties")`. Should we rather use `endsWith(...)` here, or split the path and only check for the actual file name? 
Or should we try to figure out which locations we might have used in the past to restrict it to those specifically?